xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/CoinmateStreamingService.java

```diff
@@ -0,0 +1,53 @@
+package info.bitrich.xchangestream.coinmate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.coinmate.dto.auth.AuthParams;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+import sun.rmi.runtime.Log;
+
+import java.io.IOException;
+
+/**
+ * The new API uses one WebSocket connection per channel, unlike other exchanges,
+ * which use channels over one WebSocket. This is a hack to use JsonNettyStreamingService as
+ * single channel source.
+ */
+public class CoinmateStreamingService extends JsonNettyStreamingService {
+
+  private PublishSubject<JsonNode> messages;
```

Comment:
I don't think you want or need the `PublishSubject` here. The existing channel subscription approach should work fine. Did you hit problems with it?

Reply:
You are right, channel subscription with a fake channel name works as well.

Reply:
Actually, channel subscriptions only work for the first subscriber, so I reverted to using `PublishSubject`

Reply:
As long as they are created with `share()` they should work for multiple subscribers. I have plenty of code in my application that does this.

I suspect the problem was something else.  Could you try putting up your code that uses normal channel subscriptions?  I may be able to help pinpoint the issue.